### PR TITLE
feat!: add aux chain support for merge mining

### LINKED ISSUE #5975: monero merge mining aux chain support
monero merge mining should support multiple mined chains rather than forcing tari to be the only chain.

### PR DESCRIPTION
**Pull request by SWvheerden**
Description
---
This adds aux chain support for merge mining monero

Fixes: #5975 

Motivation and Context
---
This allows tari to be merged mined with multiple other chains, up to 253 other chains. 
See: https://github.com/SChernykh/p2pool/blob/merge-mining/docs/MERGE_MINING.MD#proposed-rpc-api

How Has This Been Tested?
---

### CODE REVIEW

File: base_layer/core/src/proof_of_work/monero_rx/helpers.rs [hunk 1]
```diff
@@ -258,644 +317,657 @@ pub fn create_block_hashing_blob(
...
-        let (difficulty, hash) = get_random_x_difficulty(&input, &rx.create(&key).unwrap()).unwrap();
-        assert_eq!(
-            hash.to_hex(),
-            "f68fbc8cc85bde856cd1323e9f8e6f024483038d728835de2f8c014ff6260000"
-        );
-        assert_eq!(difficulty.as_u64(), 430603);
-    }
-}
+// #[cfg(test)]
+// mod test {
```
Comment (hansieodendaal) [inline base_layer/core/src/proof_of_work/monero_rx/helpers.rs]: I assume these tests will be updated?

Comment (SWvheerden) [reply base_layer/core/src/proof_of_work/monero_rx/helpers.rs]: fudge, good catch, I forgot I disabled those, let me fix them

File: base_layer/core/src/proof_of_work/monero_rx/helpers.rs [hunk 2]
```diff
@@ -119,15 +128,44 @@ pub fn verify_header(header: &BlockHeader) -> Result<MoneroPowData, MergeMineErr
...
 
     Ok(monero_data)
 }
 
+fn check_aux_chains(
+    monero_data: &MoneroPowData,
+    merge_mining_params: VarInt,
+    aux_chain_merkle_root: &monero::Hash,
+    tari_hash: &FixedHash,
+    gen_hash: &FixedHash,
```
Comment (hansieodendaal) [inline base_layer/core/src/proof_of_work/monero_rx/helpers.rs]:
```suggestion
    tari_genesis_block_hash: &FixedHash,
```

**Review by hansieodendaal — APPROVED** [review]
utACK

**Review by brianp — APPROVED** [review]
utAck

**Pull request merged**